### ISSUE TITLE
Image Manager pagination bug

### ISSUE THREAD

**Issue by batumibiz** [opencart/opencart#12668]
OC version - master branch on 03.09.2023

If there are more than 10 files in a single folder with images, an unpleasant pagination problem occurs.

HOW TO REPRODUCE

1. Create any folder (for example "test") in the /image/catalog/
2. Place 30-50 image files (for example, JPG) in the created folder.
3. In the admin panel, try adding an image for any product or category by selecting an image from the folder we created.
4. Look at the behavior of Image Manager.

This bug behaves differently on Linux and Windows servers (in both cases, Apache + PHP 8.1).

- On Linux, it adds several rows of images with each subsequent page.
- On Windows, the first pages are generally blank, images are visible starting from the 4th page. Further as on Linux: it adds several rows of images with each subsequent page.

Somewhere on page 7-8, all images from the folder are on the same page.
Does anyone know how to properly fix this issue?

![screenshot-isoap-2023 09 03-16_57_33](https://github.com/opencart/opencart/assets/10770779/b4540912-eb44-4198-892e-cc5dc761fb63)

**Comment by batumibiz** [comment]
I remember that version 4.0.1.1 didn't have this problem. It came later.

**Comment by ADDCreative** [comment]
@batumibiz Probably related to this issue and the attempted fix. https://github.com/opencart/opencart/issues/12259 Currently the glob pattern doesn't always match the allowed list of files.

Try the fix in the last comment to see if that helps.

**Comment by batumibiz** [comment]
> @batumibiz Probably related to this issue and the attempted fix. #12259 Currently the glob pattern doesn't always match the allowed list of files.

Thanks for the helpful advice. But unfortunately the suggested solution doesn't work either.
But still managed to solve this problem. She was in two places.

1) The first problem, as you rightly said, was glob(). But this function works unpredictably if we set GLOB_BRACE with a pattern and look for folders as well. I had to separate the search for folders and the search for files, and then merge arrays.

2) The second problem (with the list growing with each page) was in the array listing.

I fixed everything, now I will send a PR with corrections.
